docs/dev/stepbystep/guides/implement_an_instructionset.md

```diff
@@ -0,0 +1,204 @@
+# Implement an InstructionSet
+
+To support a design pattern in PatternPal one must implement the `IStepByStepRecognizer` interface. 
+This interface defines a `string` `Name`, a `RecognizerType` as defined in the protocol buffer and 
+a method `GenerateStepsList`, which will return the object required of the Step-By-Step module.
+
+# Efficiently add to `Recognizer` functionality
+
+In order to make effective use of existing `Recognizers` for both detecting patterns and Step-By-Step 
+it is recommended to encase the `IChecks` created in the `Recognizer's` `Create` method 
+(more on this method can be found [here](~/dev/recognizers/guides/implement_recognizer.md)) into separate methods. This way the checks can
+be declared once and can be called in both `Create` in `IRecognizer` and in `GenerateStepsList` in 
+`IStepByStepRecognizer`. The Recognizers would then inherit from both recognizer interfaces (`IRecognizer` 
+and `IStepByStepRecognizer`).
+
+## Prerequisites 
+
+First, it is of utmost importance that one has a clear idea of how much is asked of the user per step. It 
+is  preferred to make the steps small and that they are supported by the available `ICheck`s 
+(available checks can be found [here](~/dev/recognizers/design/checks.md)). Second, it is important to have an explicit instruction to 
+communicate to the user what is required to pass the step and optionally an explanation why the 
+requirement is the way it is. 
+
+### Requirement and explanation per step
+The goal is to make a step as small as possible. Defining a single class/interface, a single field, 
+or a single method is considered a step. Defining a method can be more detailed because the behaviour 
+of the method needs to be described (return types, modifiers, parameters), while defining a field is the 
+smallest step (including modifiers). Anything defined in one step should be closed in such a way that you
+don't have to go back the the method/interface/field in a later step. 
+
+The entity to be implemented (e.g. fields/methods/classes/interfaces) requires an explicit `Requirement` that
+states explicitly what needs to be implemented and what the underlying `ICheck` will test for. For example: 
+"Implement a static private field of the same type as the class". 
+
+In addition a `Description` of the step 
```

Comment:
Deze zin loopt niet helemaal lekker, ik zou voorstellen iets in de richting van : "For education purposes a `Description` of the step should be provided. It should explain why the step should be implemented as instructed .. "